src/map/utils/diggingutils.cpp

```diff
@@ -0,0 +1,490 @@
+			{
+				uint16 random = WELL512::irand()%1000;
+				uint8 randItem = WELL512::irand()%DigList->size();
+				if(random < (DigList->at(randItem).DigRate + map_config.dig_rate_bonus))
+				{
+					// Grab a randomly item from the list for the zone
+					// Check the requirements and then give the item if they are met
+					if(DigList->at(randItem).DigRequirements == 0)
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					if(DigList->at(randItem).DigRequirements == 1 && CanPlayerBurrow(PChar) == true) // Burrow
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					if(DigList->at(randItem).DigRequirements == 2 && CanPlayerBore(PChar) == true) // Bore
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					if(DigList->at(randItem).DigRequirements == 3 && EggHelm(PChar) == true) // Egg Helm
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					if(DigList->at(randItem).DigRequirements == 4 && NightDigging() == true) // Nighttime
+					{
+						ItemID = DigList->at(randItem).ItemID;
+					}
+					ShowDebug(CL_CYAN"Item ID Dug up %d \n", ItemID);
+				}
+				if (ItemID != 0)
+				{
+					if (PChar->getStorage(LOC_INVENTORY)->GetFreeSlotsCount() != 0)
+					{
+						// Give the item to the player
+						charutils::AddItem(PChar, LOC_INVENTORY, ItemID);
+						PChar->pushPacket(new CMessageSpecialPacket(PChar, MessageOffset3, ItemID, 0, 0));
+						UpdateZoneItemsDug(PChar);
+					}
+					else
+					{
+						// Inventory is full, throw item away
+						PChar->pushPacket(new CMessageSpecialPacket(PChar, MessageOffset, ItemID, ItemID, 0));
+					}
+				}
+				else
+				{
+					// Beat the Dig chance but not the Item chance
+					PChar->pushPacket(new CMessageTextPacket(PChar, MessageOffset2));
+				}
+			}
+			CalculateSkillUp(PChar);
+		}
+		UpdatePlayerDigCount(PChar);
+		SetLastDig(PChar);
+	}
+
+/************************************************************************
+*																		*
+*	Return the Chocobo Dig count for the character						*
+*																		*
+************************************************************************/
+
+	int8 GetPlayerDigCount(CCharEntity* PChar)
+	{
+		const int8* fmtQuery = "SELECT value FROM char_vars WHERE charid = %u AND varname = 'dig_count' LIMIT 1;";
+
+		int32 ret = Sql_Query(SqlHandle,fmtQuery,PChar->id);
+		int32 value = 0;
+
+		if (ret != SQL_ERROR &&
+			Sql_NumRows(SqlHandle) != 0 &&
+			Sql_NextRow(SqlHandle) == SQL_SUCCESS)
+		{
+			value = (int32)Sql_GetIntData(SqlHandle,0);
+		}
+		return value;
+	}
+
+/************************************************************************
+*																		*
+*	Return the character Zone In time									*
+*																		*
+************************************************************************/
+
+	int32 GetZoneInTime(CCharEntity* PChar)
+	{
+		int32 value = 0;
+
+		const int8* fmtQuery = "SELECT value FROM char_vars WHERE charid = %u AND varname = 'zoneintime' LIMIT 1;";
```

Comment:
zoneintime has nothing to do with getting more items - it, by definition, resets when you zone.  there is no reason to have it in the database